cfn-resources/federated-database-instance/cmd/resource/resource.go

```diff
@@ -241,42 +244,60 @@ func handleError(response *http.Response, method string, err error) (handler.Pro
 			Message:          fmt.Sprintf("%s:%s", method, err.Error()),
 			HandlerErrorCode: cloudformation.HandlerErrorCodeAlreadyExists}, nil
 	}
+	if response.StatusCode == http.StatusNotFound {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          fmt.Sprintf("%s:%s", method, err.Error()),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
 
 	return progress_events.GetFailedEventByResponse(fmt.Sprintf("Error during execution : %s", err.Error()),
 		response), nil
 }
 
-func (model *Model) setDataLakeTenant() (dataLakeTenant atlasSDK.DataLakeTenant) {
-	cloudProvider := constants.AWS
-	if model.DataProcessRegion.CloudProvider != nil {
-		cloudProvider = *model.DataProcessRegion.CloudProvider
-	}
-	dataLakeTenant = atlasSDK.DataLakeTenant{
-		Name: model.TenantName,
-		CloudProviderConfig: &atlasSDK.DataLakeCloudProviderConfig{
-			Aws: atlasSDK.DataLakeAWSCloudProviderConfig{
-				TestS3Bucket: *model.CloudProviderConfig.TestS3Bucket,
-				RoleId:       *model.CloudProviderConfig.RoleId,
+func (model *Model) setDataLakeTenant() (dataLakeTenant admin.DataLakeTenant) {
+	dataLakeTenant = admin.DataLakeTenant{
+		Name:    model.TenantName,
+		Storage: model.newDataFederationDataStorage(),
+	}
+
+	cloudProviderConfig := model.CloudProviderConfig
+	if cloudProviderConfig != nil && cloudProviderConfig.TestS3Bucket != nil && cloudProviderConfig.RoleId != nil {
+		dataLakeTenant.CloudProviderConfig = &admin.DataLakeCloudProviderConfig{
+			Aws: admin.DataLakeAWSCloudProviderConfig{
+				TestS3Bucket: *cloudProviderConfig.TestS3Bucket,
+				RoleId:       *cloudProviderConfig.RoleId,
 			},
-		},
-		DataProcessRegion: &atlasSDK.DataLakeDataProcessRegion{
-			Region:        *model.DataProcessRegion.Region,
-			CloudProvider: cloudProvider,
-		},
+		}
+	}
 
-		Storage: model.newDataFederationDataStorage(),
+	dataProcessRegion := model.DataProcessRegion
+	if dataProcessRegion != nil && dataProcessRegion.CloudProvider != nil && dataProcessRegion.Region != nil {
+		dataLakeTenant.DataProcessRegion.CloudProvider = *dataProcessRegion.CloudProvider
+		dataLakeTenant.DataProcessRegion = &admin.DataLakeDataProcessRegion{
+			CloudProvider: *dataProcessRegion.CloudProvider,
+			Region:        *dataProcessRegion.Region,
+		}
 	}
+
+	if dataProcessRegion != nil && dataProcessRegion.Region != nil {
+		dataLakeTenant.DataProcessRegion = &admin.DataLakeDataProcessRegion{
```

Comment:
Can we have the scenario where a users defines `DataProcessRegion` with only the `CloudProvider` defined? Would we want to still define data process region in this case or was this causing issues?

Reply:
For the API, when `DataProcessRegion` is defined both `Region` and `CloudProvider` are required. In the case of CFN the default is AWS so that means that in this case `Region` can be specified without specifying `CloudProvider`. This if is handling this specific edge case of CFN

Reply:
> where `CloudProvider` is defined but not `Region`
As per the API this should not be possible, both are required. I see that the current implementation would ignore `DataProcessRegion` all together in that case. I would not put it as required. 
Question is how we want to handle this case: ignore `DataProcessRegion`(as it is now) or throw an error? Note that his method is called from both Create and Update operations.

Reply:
I would think from the users point of view, what documentation or error messages are in place to understand that if `DataProcessRegion` is defined, then `Region` must be defined and `CloudProvider` is optional as it has the default value? With this in mind we can adjust if needed.

Reply:
I would say that these 2 docs ( [first](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/federated-database-instance/docs/README.md#dataprocessregion), [second](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/cfn-resources/federated-database-instance/docs/dataprocessregion.md)) should be enough. Also it explicitly says that only supported cloud provider is AWS. WDYT?

Reply:
Adressed making Region required when DataProcessRegion is defined